src/FormControl/FormControl.tsx

```diff
@@ -0,0 +1,232 @@
+  )
+  const inputProps = React.isValidElement(InputComponent) ? InputComponent.props : undefined
+
+  if (!InputComponent) {
+    // eslint-disable-next-line no-console
+    console.warn(
+      `To correctly render this field with the correct ARIA attributes passed to the input, please pass one of the component from @primer/react as a direct child of the FormControl component: ${expectedInputComponents.reduce(
+        (acc, componentName) => {
+          acc += `\n- ${componentName.displayName}`
+
+          return acc
+        },
+        ''
+      )}`,
+      'If you are using a custom input component, please be sure to follow WCAG guidelines to make your form control accessible.'
+    )
+  } else {
+    if (inputProps?.id) {
+      // eslint-disable-next-line no-console
+      console.warn(
+        `instead of passing the 'id' prop directly to the input component, it should be passed to the parent component, <FormControl>`
+      )
+    }
+    if (inputProps?.disabled) {
+      // eslint-disable-next-line no-console
+      console.warn(
+        `instead of passing the 'disabled' prop directly to the input component, it should be passed to the parent component, <FormControl>`
+      )
+    }
+    if (inputProps?.required) {
+      // eslint-disable-next-line no-console
+      console.warn(
+        `instead of passing the 'required' prop directly to the input component, it should be passed to the parent component, <FormControl>`
+      )
+    }
+  }
+
+  if (!labelChild) {
+    // eslint-disable-next-line no-console
+    console.error(
+      `The input field with the id ${id} MUST have a FormControl.Label child.\n\nIf you want to hide the label, pass the 'visuallyHidden' prop to the FormControl.Label component.`
+    )
+  }
+
+  if (variant === 'choice') {
+    if (validationChild) {
+      // eslint-disable-next-line no-console
+      console.warn(
+        'Validation messages are not rendered for an individual checkbox or radio. The validation message should be shown for all options.'
+      )
+    }
+
+    if (React.Children.toArray(children).find(child => React.isValidElement(child) && child.props?.required)) {
+      // eslint-disable-next-line no-console
+      console.warn('An individual checkbox or radio cannot be a required field.')
+    }
+  } else {
+    // TODO: reconsider if we even need this check. The UI will just look like something is wrong
+    if (React.isValidElement(InputComponent) && (InputComponent.type === Checkbox || InputComponent.type === Radio)) {
+      // eslint-disable-next-line no-console
+      console.warn('The Checkbox or Radio components are only intended to be used with the "choice" variant')
+    }
+
+    if (
+      React.Children.toArray(children).find(
+        child => React.isValidElement(child) && child.type === FormControlLeadingVisual
+      )
+    ) {
+      // eslint-disable-next-line no-console
+      console.warn(
```

Comment:
Jest specifically tests to make sure these are logged, so I don't think we can.

Reply:
Rather than testing the global, can you do something like: 

```js

const mockWarningFn = jest.fn();

beforeAll(() => {
  jest.spyOn(global.console, 'warn').mockImplementation(mockFn);
});

afterAll(() => {
  jest.clearAllMocks();
});


...

// where you're checking it...
expect(mockWarningFn).toHaveBeenCalled();
```